src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java

```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
         CaseJudgementUpdate.updateCaseWithJudgement(caseData, judgementType);
     }
 
-    private void updateRespondentRep(CaseData caseData, RepresentedTypeR representedType) {
-
-        //case 1: update is adding
-
-        //case 2: update is removing
+    private void updateRespondentRep(CaseData caseData, UpdateDataModel updateDataModel) {
+        RepresentedTypeR representedType = updateDataModel.getRepresentedType();
+        String isRespondentRepRemovalUpdate = updateDataModel.getIsRespondentRepRemovalUpdate();
 
         if (caseData.getRespondentCollection() != null) {
+            boolean respondentRepRemovalUpdate = isRespondentRepRemovalUpdate.equals(YES);
+
             Optional<RespondentSumTypeItem> respondentSumTypeItemOptional =
                     caseData.getRespondentCollection().stream()
                             .filter(respondentSumTypeItem ->
                                     respondentSumTypeItem.getValue().getRespondentName()
                                             .equals(representedType.getRespRepName()))
                             .findAny();
 
-            if (respondentSumTypeItemOptional.isPresent()) {
+            boolean respondentSumTypeIsPresent = respondentSumTypeItemOptional.isPresent();
 
-                if (caseData.getRepCollection() != null) {
-
-                    boolean found = false;
+            if(respondentSumTypeIsPresent){
```

Comment:
Minor thing but you might as well write
`if (respondentSumTypeItemOptional.isPresent()) {`